roles/cockpit/templates/cockpit.yml.j2

```diff
@@ -0,0 +1,25 @@
+http:
+  routers:
+    {{ cockpit_name }}-http:
+      entryPoints:
+        - "web"
+      rule: "Host(`{{ cockpit_web_subdomain + '.' + cockpit_web_domain }}`)"
+      middlewares:
+        {{ traefik_default_middleware_http.split(',') | to_nice_yaml | trim | indent(8) }}
+      service: "{{ cockpit_name }}"
+    {{ cockpit_name }}:
+      entryPoints:
+        - "websecure"
+      rule: "Host(`{{ cockpit_web_subdomain + '.' + cockpit_web_domain }}`)"
```

Comment:
Maybe try using traefik_host here as it should do the empty subdomain check for you.

Reply:
Is that defined somewhere? if subdomain isn't defined it'll start with `.` which ain't no good. but I'm not sure, should I be defining traefik_host, or its done by the role somehow I'm unaware of?

Reply:
Yes, it is a global variable.